vault.js

```diff
@@ -1,4 +1,50 @@
+/*jshint esversion: 6 */
 'use strict';
-module.exports = function() {
-  
-};
\ No newline at end of file
+function vaultGenerator() {
+   let obj = {};
+  function setValue(key, value){
+
+      obj[key]= value;
+      return obj;
+  }
+  function getValue(key){
+    //if no key is used, return null
+    if(!key){
```

Comment:
This can be simplified, needing less checks.

There is an edge case this solution does not cover.
```
const vault = vaultGenerator();
vault.getValue('toString'); // [Function toString]
// should return null
```